libraries/grpc-sdk/src/helpers/wrapRouterFunctions.ts

```diff
@@ -68,14 +68,23 @@ export function wrapRouterGrpcFunction(
           if (typeof r === 'string') {
             callback(null, { result: r });
           } else {
+            const result = r.result ? JSON.stringify(r.result) : undefined;
```

Comment:
Do not modify the order in which parsing and checks happen. Add an "if" statement on the top, when you create a "response" object that contains the cookie options. After that the rest of the checks should add their data in that object. For example, line 69 would do: callback(null, {...responseObject,result:r}